tests/test_offshore_inputs.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+Offshore Inputs tests
+"""
+from click.testing import CliRunner
+import json
+import os
+import pandas as pd
+from pandas.testing import assert_frame_equal
+import pytest
+import tempfile
+import traceback
+
+from rex.utilities.loggers import LOGGERS
+from reVX import TESTDATADIR
+from reVX.offshore.offshore_inputs import OffshoreInputs
+from reVX.offshore.offshore_inputs_cli import main
+
+INPUTS_FPATH = os.path.join(TESTDATADIR, 'offshore', 'dist_to_coast.h5')
+OFFSHORE_SITES = os.path.join(TESTDATADIR, 'wtk', 'ri_100_wtk_2012.h5')
+BASELINE = os.path.join(TESTDATADIR, 'offshore', 'inputs_baseline.csv')
+INPUT_LAYERS = {'dist_to_coast': 'dist_s_to_l',
+                'ports_operations': 'dist_op_to_s'}
+
+
+def test_offshore_inputs():
+    """
+    Compute offshore inputs
+    """
+    baseline = pd.read_csv(BASELINE)
+    test = OffshoreInputs.extract(INPUTS_FPATH, OFFSHORE_SITES, INPUT_LAYERS)
+
+    assert_frame_equal(baseline, test, check_dtype=False)
+
+
+@pytest.fixture(scope="module")
+def runner():
+    """
+    cli runner
+    """
+    return CliRunner()
+
+
+def test_cli(runner):
+    """
+    Test CLI
+    """
+
+    with tempfile.TemporaryDirectory() as td:
+        input_layers = os.path.join(td, 'input_layers.json')
+        with open(input_layers, 'w') as f:
+            json.dump({'input_layers': INPUT_LAYERS}, f)
+
+        config = {
+            "directories": {
+                "log_directory": td,
+                "output_directory": td
+            },
+            "execution_control": {
+                "option": "local"
+            },
+            "inputs_fpath": INPUTS_FPATH,
+            "offshore_sites": OFFSHORE_SITES,
+            "input_layers": input_layers,
+        }
+        config_path = os.path.join(td, 'config.json')
+        with open(config_path, 'w') as f:
+            json.dump(config, f)
+
+        result = runner.invoke(main, ['from-config',
+                                      '-c', config_path])
+        msg = ('Failed with error {}'
+               .format(traceback.print_exception(*result.exc_info)))
+        assert result.exit_code == 0, msg
+
+        test = os.path.basename(input_layers).replace('.json', '.csv')
+        test = os.path.join(td, test)
+        test = pd.read_csv(test)
+
+        baseline = pd.read_csv(BASELINE)
+        assert_frame_equal(baseline, test, check_dtype=False)
```

Comment:
I think real checks would be valuable here. For example: check the offshore meta data and go manually retrieve the row/col indices from the h5 and assert equals. Check that the offshore gid to exclusion h5 index is done correctly.

Reply:
Good call